src/kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs

```diff
@@ -368,22 +382,16 @@ impl<'tcx> GotocCtx<'tcx> {
                     );
                 }
 
-                if destination.is_none() {
-                    // No target block means this function doesn't return.
-                    // This should have been handled by the Nevers hook.
-                    return self.codegen_assert_false(
-                        PropertyClass::SanityCheck,
-                        &format!("reach some nonterminating function: {:?}", func),
-                        loc.clone(),
-                    );
-                }
-
-                let (p, target) = destination.unwrap();
+                let (place, target) =
+                    if let Some((p, t)) = destination { (Some(p), Some(t)) } else { (None, None) };
```

Comment:
Nit: This boggled me. Can we convert to match instead of if let?

Reply:
I can. But why did it bother you?

Reply:
I think my brain treats `if let` as statement-like and `match` as expression-like. This is despite being happy with `if` as expression-like.

It's possible it's just me needing to get used to it. If you disagree, no problem.

Reply:
It took me some time to get used to as well :)

Reply:
If you don't mind, I prefer to move on with this PR since there's nothing else to change.

I don't have any strong opinion about this. I tend to favor `if let` when I'm looking for one specific match. It's less verbose and clear to me that it's an all or nothing situation. We can discuss this later and if one is preferred over the other, maybe we can have a little convention page.

Reply:
Yeah, that's fine, I prefer velocity over this nit.

Reply:
Thanks